includes/functions.php

```diff
@@ -279,6 +279,20 @@ function is_activitypub_request() {
 		return false;
 	}
 
+	// Check if the current post type supports ActivityPub.
+	if ( \is_singular() ) {
+		$queried_object = \get_queried_object();
+		$post_type      = \get_post_type( $queried_object );
+		$post_types     = \get_post_types_by_support( 'activitypub' );
+
+		if (
+			false === $post_type ||
+			! \in_array( $post_type, $post_types, true )
+		) {
+			return false;
+		}
```

Comment:
Once you have the post type, I think you may be able to rely on `post_type_supports`:

```suggestion
		return ! empty( $post_type ) && \post_type_supports( $post_type, 'activitypub' );
```

Reply:
thanks for the feedback, but I still have to rewrite it a bit, to only return on `false`.